File: python/rikai/spark/sql/model_resolver.py
from pyspark.sql import SparkSession


class ModelResolver(object):
    def __init__(self, spark: SparkSession):
        self.spark = spark

    def __repr__(self):
        return "ModelResolver"

    def resolve(self, uri: str):
        pass

    def register(self):
        jvm = self.spark.sparkContext._jvm
        print("I am calling you")
        v = jvm.ai.eto.rikai.sql.spark.ModelResolverHolder.register(self)
        print(f"You telling me {v}")

    def toString(self):
        return repr(self)

    class Java:
        implements = ["ai.eto.rikai.sql.spark.ModelResolver"]

File: src/main/scala/ai/eto/rikai/sql/spark/ModelResolver.scala
package ai.eto.rikai.sql.spark

import ai.eto.rikai.sql.model.Model

/**
  * ModelResolver is an interface used in python
  */
trait ModelResolver {

  def resolve(uri: String): Model
}

object ModelResolverHolder {
  var resolver: ModelResolver = null

  def register(mr: ModelResolver): String = {
    println(s"XXXXXXXXX\n\n\nHey I registerd ${mr}")
    resolver = mr
    "fick"
  }
}
